keyboards/handwired/dactyl_manuform/4x5/4x5.h

```diff
@@ -20,12 +20,12 @@
     { XXX, L31, L32, L33, L34 }, \
     { XXX, L41, L42, L43, L44 }, \
 \
-    { R04, R03, R02, R01, R00 }, \
-    { R14, R13, R12, R11, R10 }, \
-    { R24, R23, R22, R21, R20 }, \
-    { XXX, R33, R32, R31, R30 }, \
-    { XXX, R43, R42, R41, R40 } \
-}
+    { R00, R01, R02, R03, R04 }, \
```

Comment:
You can't just flip the order here without changing the order above. 

As is, this will flip the half.

Reply:
This won’t flip the half if you follow the most popular wiring scheme of dactyl manuform on the internet (e.g. from tshort’s repo). Pay attention to the order of the col pin and the connected columns on the right hand, it’s different from the left. I wired my keyboards like that, and flashed the original version here, then the keys flipped on the right. More importantly, the defined XXX positions actually disabled two thumb keys on the right hand side. 
Anyway, the most important thing is the order in parentheses **must be** consistent with the order in the brackets. Otherwise, the firmware will flip the order, because the “physical ” coordinate of the key is represented by the orders in the brackets. For example, according to the original order, if I define the top-left key is “q”, it assigns q to R04 according to the mapping in the bracket. Then, if I hit the top-left key, it won’t be q; instead, the top-right key will be the “q”, as it is assigned as “R04” in the parentheses. The order of mapping becomes inconsistent.

I have tested this modified version on my two latest build dactyl manuforms, which are both wired according to the mentioned scheme. They both works without any incorrect flipping or unexpected key-disabling.
![](https://github.com/tshort/dactyl-keyboard/blob/master/resources/dactyl_manuform_right_wire_diagram.png)
)